src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -79,6 +90,38 @@ protected function maybeRenderId(Component\Component $component, $tpl) {
 		}
 	}
 
+	protected function renderMonth(Component\Button\Month $component, RendererInterface $default_renderer) {
+		$def = $component->getDefault();
+
+		for ($i = 1; $i<=12; $i++)
+		{
+			$this->toJS(array("month_".str_pad($i, 2, "0", STR_PAD_LEFT)."_long"));
+		}
+
+		$tpl = $this->getTemplate("tpl.month.html", true, true);
+
+		$month = explode("-", $def);
+		$tpl->setVariable("DEFAULT_LABEL", $this->txt("month_".str_pad($month[0], 2, "0", STR_PAD_LEFT)."_long")." ".$month[1]);
+
+		include_once("./Services/Calendar/classes/class.ilCalendarUtil.php");
+		\ilCalendarUtil::initDateTimePicker();
```

Comment:
I agree that this line is an issue. I thought this was ok, since Services/Calendar is currently responsible for managing the picker.
> Load those files here directly.
What would we win? We still have the dependency to Services/Calendar and now added redundancy, since two php code files would load the same js files.

 > This later part I would consider optional

Even if we move it, there still should be some php wrapper being responsible to handle it, right? Currently this is responsibility of Services/Calendar, and there are dependencies in "the old" form implementation to this service. If we want to get rid of the Services/Calendar dependency, we would need to refactor this completely and create a dependency from the old form implementation to the new ui framework instead. I think this will be much easier as soon as we have a form implementation in the new UI framework. I did not expect that this would be necessary at the current state.